experimental/framework/frs-client/src/FrsClient.ts

```diff
@@ -0,0 +1,176 @@
+} from "./interfaces";
+import { FrsAudience } from "./FrsAudience";
+import { FrsSimpleUrlResolver } from "./FrsSimpleUrlResolver";
+
+/**
+ * FrsClientInstance provides the ability to have a Fluid object backed by the FRS service
+ */
+export class FrsClientInstance {
+    public readonly documentServiceFactory: IDocumentServiceFactory;
+    public readonly user;
+    constructor(private readonly connectionConfig: FrsConnectionConfig) {
+        this.user = this.connectionConfig.user
+            ? { id: this.connectionConfig.user.userId, name: this.connectionConfig.user.userName }
+            : generateUser();
+        const tokenProvider: ITokenProvider = connectionConfig.tokenProvider
+            ?? new InsecureTokenProvider(connectionConfig.key, this.user);
+        this.documentServiceFactory = new RouterliciousDocumentServiceFactory(
+            tokenProvider,
+        );
+    }
+
+    public async createContainer(
+        containerConfig: FrsContainerConfig,
+        containerSchema: ContainerSchema,
+    ): Promise<[FluidContainer, FrsContainerServices]> {
+        return this.getContainerCore(
+            containerConfig,
+            containerSchema,
+            true,
+        );
+    }
+
+    public async getContainer(
+        containerConfig: FrsContainerConfig,
+        containerSchema: ContainerSchema,
+    ): Promise<[FluidContainer, FrsContainerServices]> {
+        return this.getContainerCore(
+            containerConfig,
+            containerSchema,
+            false,
+        );
+    }
+
+    private async getFluidContainerAndServices(
+        container: Container,
+    ): Promise<[FluidContainer, FrsContainerServices]> {
+        const rootDataObject = await requestFluidObject<RootDataObject>(container, "/");
+        const fluidContainer = new FluidContainer(container, rootDataObject);
+        const containerServices = this.getContainerServices(container);
+        return [fluidContainer, containerServices];
+    }
+
+    private getContainerServices(
+        container: Container,
+    ): FrsContainerServices {
+        return {
+            audience: new FrsAudience(container),
+        };
+    }
+
+    private async getContainerCore(
+        containerConfig: FrsContainerConfig,
+        containerSchema: ContainerSchema,
+        createNew: boolean,
+    ): Promise<[FluidContainer, FrsContainerServices]> {
+        const runtimeFactory = new DOProviderContainerRuntimeFactory(
+            containerSchema,
+        );
+        const module = { fluidExport: runtimeFactory };
+        const codeLoader = { load: async () => module };
+        const urlResolver = new FrsSimpleUrlResolver(this.connectionConfig, containerConfig.id);
+        const loader = new Loader({
+            urlResolver,
+            documentServiceFactory: this.documentServiceFactory,
+            codeLoader,
+            logger: containerConfig.logger,
+        });
+
+        let container: Container;
+
+        if (createNew) {
```

Comment:
The shared code here is primarily just making the `Loader`, so I'd suggest using a `makeLoader()` rather than `getContainerCore()` and pulling the new/existing logic up into `createContainer()`/`getContainer()`.  This will let you avoid the split->merge->re-split code flow.

Reply:
Refactored the code as suggested! Thanks!! 🙇